src/main.cpp

```diff
@@ -2080,6 +2081,47 @@ void PartitionCheck(bool (*initialDownloadCheck)(), CCriticalSection& cs, const
     }
 }
 
+Consensus::VersionBitsConditionChecker checkCSV(Consensus::DEPLOYMENT_CSV);
```

Comment:
These could be static and be in consensus/versionbits.cpp instead of main.cpp by moving ComputeBlockVersion() to versionbits as well (of course, without cs_main).

WarningBitsConditionChecker could be encapsulated behind a function there too. In a very far future in which Bitcoin Core only uses libconsensus through its C API, that ```GetVersionBitsWarnings(const CBlockIndex* pindexPrev, const Consensus::Params&)``` function seems much closer from being "exposable" in libconsensus than class ```Consensus::AbstractThresholdConditionChecker```.

I know that ComputeBlockVersion() and GetVersionBitsWarnings() are not strictly consensus code, but they heavily depend on consensus code, and I think it would be a wasted opportunity not to have this functionality as encapsulated as possible from birth.